posthog/warehouse/data_load/test/test_service.py

```diff
@@ -0,0 +1,69 @@
+import unittest
+from datetime import timedelta
+from temporalio.client import ScheduleCalendarSpec, ScheduleRange
+
+from posthog.warehouse.data_load.service import get_calendar_spec
+
+
+class TestGetCalendarSpec(unittest.TestCase):
+    def test_basic_daily_schedule(self):
+        # Test with daily schedule at midnight, expect 1 execution per day
+        result = get_calendar_spec(0, 0, timedelta(days=1))
+        expected = ScheduleCalendarSpec(
+            hour=[ScheduleRange(start=0, end=0, step=24)], minute=[ScheduleRange(start=0, end=0, step=1)]
+        )
+        self.assertEqual(result.hour, expected.hour)
+        self.assertEqual(result.minute, expected.minute)
+
+    def test_basic_daily_schedule_at_12_am(self):
+        # Test with daily schedule at 12:00, expect 4 executions per day
+        result = get_calendar_spec(0, 0, timedelta(hours=6))
+        expected = ScheduleCalendarSpec(
+            hour=[ScheduleRange(start=0, end=18, step=6)], minute=[ScheduleRange(start=0, end=0, step=1)]
+        )
+        self.assertEqual(result.hour, expected.hour)
+        self.assertEqual(result.minute, expected.minute)
+
+    def test_every_6_hours(self):
+        # Test with 6-hour frequency starting at 3:00, expect 4 executions per day
+        result = get_calendar_spec(3, 0, timedelta(hours=6))
+        expected = ScheduleCalendarSpec(
+            hour=[ScheduleRange(start=3, end=21, step=6)], minute=[ScheduleRange(start=0, end=0, step=1)]
+        )
+        self.assertEqual(result.hour, expected.hour)
+        self.assertEqual(result.minute, expected.minute)
+
+    def test_every_12_hours(self):
+        # Test with 12-hour frequency starting at 6:30, expect 2 executions per day
+        result = get_calendar_spec(6, 30, timedelta(hours=12))
+        expected = ScheduleCalendarSpec(
+            hour=[ScheduleRange(start=6, end=18, step=12)], minute=[ScheduleRange(start=30, end=30, step=1)]
+        )
+        self.assertEqual(result.hour, expected.hour)
+        self.assertEqual(result.minute, expected.minute)
+
+    def test_odd_hour_frequency(self):
+        # Test with 5-hour frequency starting at 2:15
+        result = get_calendar_spec(2, 15, timedelta(hours=5))
+        expected = ScheduleCalendarSpec(
+            hour=[ScheduleRange(start=2, end=22, step=5)], minute=[ScheduleRange(start=15, end=15, step=1)]
+        )
+        self.assertEqual(result.hour, expected.hour)
+        self.assertEqual(result.minute, expected.minute)
```

Comment:
My concern with this is that there will be 5-hours between each interval, except for when it crosses midnight - then the last interval is only 4 hours:
- 02:15 AM
- 07:15 AM
- 12:15 PM
- 05:15 PM
- 10:15 PM
- 02:15 AM

Is this what we want? I'm not so sure what the expectation would be from the users POV. Running something actually every 5 hours starting at 2:15am can be achieved with:
```python
ScheduleIntervalSpec(
    every=timedelta(hours=5),  # Every 5 hours
    offset=timedelta(hours=2, minutes=15)  # Start at 2:15am
)
```
This would give something like:
- 02:15 AM
- 07:15 AM
- 12:15 PM
- 05:15 PM
- 10:15 PM
- 03:15 AM
- 08:15 AM
- 01:15 PM
- 06:15 PM
- 11:15 PM

Reply:
I think I would expect the anchor time to be respected always and have the inconsistent gap... But I suppose the practical reality of this is that we would never allow an interval that is not a factor of 24 to avoid this kind of thing. So in a practical sense, it wouldn't really matter, but the interval time sure is simpler to understand configuration wise.

Reply:
I tried to change this over to the much simpler config you suggested, but I ran into this issue where the offset cannot be greater than the sync_frequency which makes something like start at `7:30 AM` and sync every 6 hours throw an exception. I think as a result, I'll keep it like it is if that's alright.

```
Invalid schedule spec: phase cannot be greater than Interval
```

Reply:
Ahh gotcha. I guess in that case we should see the offset to `7:30 - 6 hours`. But not something we need to worry about now - can follow up later if we want - good stuff on this!